server.js

```diff
@@ -32,9 +32,30 @@ const { createState } = require("./utils/state")
 // set the view engine to ejs
 app.set('view engine', 'ejs')
 
+app.use('/public/', express.static('./public'));
+
+// Load theme data
+let themeData;
+const theme_endpoint = `${config.api.environment}/a/consumer/api/v0/institutions/${config.api.institutionId}/themes`
+fetch(theme_endpoint, { method: 'get' })
+    .then(res => res.json())
+    .then(json => {
+        themeData = json;
+    })
+    .catch((err) => {
+        console.log("Unable to fetch theme data - ", err)
+    })
```

Comment:
**issue:** We probably shouldn't encourage folks to dynamically update their plugin's theme based on the theme configured for the financial institution's Banno environment.

While it seems fairly logical to do (e.g. to always be up-to-date if/when an FI updates their them, or for fintechs which want to have a single codebase that operates across different FIs), the real-world impact is that plugin authors end up with heavy usage of their plugin and they ask the Developer Relations team on how they can improve performance.

In general, we tell folks to heavily pre-calculate or cache anything that they can do, such a this excerpt from the [Designing and Developing Plugins](https://jackhenry.dev/open-api-docs/plugins/guides/designinganddevelopingplugins/#caching) guide:

> Plugins appear on the user’s dashboard in Banno. The dashboard is the default screen shown to the user when they log into Banno Online and Banno Mobile. As such, it is the most frequently viewed screen across all of Banno.
>
> This means that the Dashboard will drive a high volume of traffic to your plugin via the card face. Recall that the card face should quickly provide users with the most important and most frequently used information, at a glance.
>
> When possible, it is recommended to pre-calculate and/or cache data for the card face to enable a better user experience and handle the volume of traffic that will be sent to your plugin and your web server.

Following along those same lines, the [Plugin theming](https://jackhenry.dev/open-api-docs/plugins/guides/plugintheming/#using-the-theme-data-in-your-plugin) guide recommends that folks fetch the theme data _offline_ and use that configuration as-needed for their plugin. It ends up being the most practical way to have a configuration-driven yet highly performant plugin:

Financial institutions typically do not update their themes very often, because of this, we do not recommend fetching theme data in real-time for your plugin. It is advised to store this theme data and only update it at as necessary.

> Within your plugin, identify which design elements should be customized for each financial institution. This customization could range from altering the background or text color to more complex changes.
> 
> The method of implementing these customizable styles depends on your development environment. However, it is advisable to utilize semantic class names, such as bg-primary or text-primary, for styling. This approach not only ensures consistency in design but also simplifies the application of themes by clearly defining the purpose of each color within the interface.

Reply:
I can pull that theme logic out, but before I do, I wanted to mention that this call is only done **once, when the node app starts up**. The app does not make that API call again while it's up and serving requests. 

So the app logic looks like this:
- The app starts up on its server
- The app pulls the theme data, one time only
- Now the app is then ready to serve requests:
    - A request comes in -> the app responds (repeat)

I'm totally cool with removing that theme logic, but I'm wondering if knowing that this 1-second API call that happens **only once, before any requests are served** might change your thoughts on this. I'm good either way.

Reply:
I appreciate the thoughts and feedback.

However, I'd still like to pull the dynamic nature of the API-pull of theme data (which can instead be a local config/code...with commentary about folks needing per-FI configs if they serve more than one FI).

The reasoning is that folks often, despite our guidance (and my own personal protests), end up using these example projects almost as-is for their production deployments.

We always caveat with 'this is educational code, not Production code' yet folks will still reach out to us with _'Hey Jack Henry, our plugin is doing bad stuff in (Scenario), when will you fix as this is holding up our project'_ and we have to tell them (again) that the Simple Plugin Example (and its siblings) are _educational_ code for learning purposes, not fork-and-deploy-to-Prod ready-made projects.

Reply:
OK makes sense, got that. I'll remove that theme logic, I'll freeze those light/dark mode CSS values into a separate CSS file that both static.ejs and dynamic.ejs share, then I'll resubmit (probably tomorrow).